numpy/core/src/multiarray/dtype_traversal.h

```diff
@@ -0,0 +1,105 @@
+#ifndef NUMPY_CORE_SRC_MULTIARRAY_DTYPE_TRAVERSAL_H_
+#define NUMPY_CORE_SRC_MULTIARRAY_DTYPE_TRAVERSAL_H_
+
+#include "array_method.h"
+
+/*
+ * A simplified loop, similar to a general strided-loop function.
+ * Using a `void *reserved`, because I think we probably need to pass in
+ * Intepreter state or similar in the future.  But I don't want to pass in
+ * a full context (with pointers to dtypes, method, caller which all make
+ * no sense for a simple function).
+ *
+ * We assume for now that this context can be just passed through in the
+ * the future (for structured dtypes).
+ */
+typedef int (simple_loop_function)(
```

Comment:
nit: maybe these two new function signatures should be called `traverse_loop_function` and `get_traverse_loop_function` to indicate that it's a loop specifically for array traversal

Reply:
Fair point, "simple" is always a bad name anyway ;).